stdlib/REPL/test/repl.jl

```diff
@@ -1833,23 +1833,50 @@ fake_repl() do stdin_write, stdout_read, repl
     @test contains(txt, "Some type information was truncated. Use `show(err)` to see complete types.")
 end
 
-try # test the functionality of `UndefVarError_hint` against `Base.remove_linenums!`
+try # test the functionality of `UndefVarError_hint`
     @assert isempty(Base.Experimental._hint_handlers)
     Base.Experimental.register_error_hint(REPL.UndefVarError_hint, UndefVarError)
 
-    # check the requirement to trigger the hint via `UndefVarError_hint`
-    @test !isdefined(Main, :remove_linenums!) && Base.ispublic(Base, :remove_linenums!)
-
     fake_repl() do stdin_write, stdout_read, repl
         backend = REPL.REPLBackend()
         repltask = @async REPL.run_repl(repl; backend)
-        write(stdin_write,
-              "remove_linenums!\n\"ZZZZZ\"\n")
+        write(stdin_write, """
+        module AAAA # can't clash with anything else (module A defined elsewhere)
+            export f
+            f() = 0.0
+        end
+
+        module C_outer
+            import ..AAAA: f
```

Comment:
An additional important case to test is that of available but unresolved bindings, e.g.
```julia
julia> module AAAA
           export f
           f() = 0.0
       end
Main.AAAA

julia> module B
           using ..AAAA
           export g, f
           g(x) = 69x + 420
       end
Main.B

julia> Base.isbindingresolved(B, :f)
false

julia> isdefined(B, :f)  # note that `isdefined` forces binding resolution
true
```

Reply:
Interesting, thank you.  What is the best place to find docs on this?  Is getting the desired behavior just a matter of changing `Base.isbindingresolved` to `isdefined` in line 81?
```julia
        if !Base.isbindingresolved(m, var) || (!Base.isexported(m, var) && !Base.ispublic(m, var))
```